airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcTargetPosition.java

```diff
@@ -105,10 +109,27 @@ public boolean isEventAheadOffset(final Map<String, String> offset, final Change
 
     final JsonNode offsetJson = Jsons.deserialize((String) offset.values().toArray()[0]);
 
-    final String offset_lsn =
-        offsetJson.get("lsn_commit") != null ? String.valueOf(offsetJson.get("lsn_commit")) : String.valueOf(offsetJson.get("lsn"));
-    final String event_lsn = String.valueOf(event.eventValueAsJson().get("source").get("lsn"));
-    return Long.parseLong(event_lsn) > Long.parseLong(offset_lsn);
+    final String stateOffsetLsnCommit = offsetJson.get("lsn_commit") != null ? String.valueOf(offsetJson.get("lsn_commit")) : null;
+    if (stateOffsetLsnCommit == null) {
+      return false;
+    }
+
+    try {
```

Comment:
Can you add a comment with a typical structure of the event you're parsing. To make it easier to uderstand

Reply:
NIT:
```
if (offsetJson.get("lsn_commit") == null) {
  return false;
}
final stateOffsetLsnCommit =String.valueOf(offsetJson.get("lsn_commit")); 
```